src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8763,6 +8769,41 @@ private static Precedence GetPrecedence(SyntaxKind op)
                     return Precedence.Range;
                 case SyntaxKind.ConditionalExpression:
                     return Precedence.Expression;
+                case SyntaxKind.AliasQualifiedName:
+                case SyntaxKind.AnonymousObjectCreationExpression:
+                case SyntaxKind.ArgListExpression:
+                case SyntaxKind.ArrayCreationExpression:
+                case SyntaxKind.BaseExpression:
+                case SyntaxKind.CharacterLiteralExpression:
+                case SyntaxKind.ConditionalAccessExpression:
+                case SyntaxKind.DeclarationExpression:
+                case SyntaxKind.DefaultExpression:
+                case SyntaxKind.DefaultLiteralExpression:
+                case SyntaxKind.ElementAccessExpression:
+                case SyntaxKind.FalseLiteralExpression:
+                case SyntaxKind.GenericName:
+                case SyntaxKind.IdentifierName:
+                case SyntaxKind.ImplicitArrayCreationExpression:
+                case SyntaxKind.ImplicitStackAllocArrayCreationExpression:
+                case SyntaxKind.InterpolatedStringExpression:
+                case SyntaxKind.InvocationExpression:
+                case SyntaxKind.NullLiteralExpression:
+                case SyntaxKind.NumericLiteralExpression:
+                case SyntaxKind.ObjectCreationExpression:
+                case SyntaxKind.ParenthesizedExpression:
+                case SyntaxKind.PointerMemberAccessExpression:
+                case SyntaxKind.PostDecrementExpression:
+                case SyntaxKind.PostIncrementExpression:
+                case SyntaxKind.PredefinedType:
+                case SyntaxKind.RefExpression:
+                case SyntaxKind.SimpleMemberAccessExpression:
+                case SyntaxKind.StackAllocArrayCreationExpression:
+                case SyntaxKind.StringLiteralExpression:
+                case SyntaxKind.SuppressNullableWarningExpression:
+                case SyntaxKind.ThisExpression:
+                case SyntaxKind.TrueLiteralExpression:
+                case SyntaxKind.TupleExpression:
+                    return Precedence.Primary;
                 default:
                     throw ExceptionUtilities.UnexpectedValue(op);
```

Comment:
>throw ExceptionUtilities.UnexpectedValue(op); [](start = 20, length = 45)

Are we calling `GetPrecedence()` for more (all?) expression types now? If so, are we likely to hit this for new expression types?

Reply:
Yes, we do.  On line 9048.

---
In reply to: [352907086](https://github.com/dotnet/roslyn/pull/40089#discussion_r352907086) [](ancestors = 352907086)

Reply:
In that case, how do we test all future expression types to ensure the associated `SyntaxKind` is added above?

---
In reply to: [353406709](https://github.com/dotnet/roslyn/pull/40089#discussion_r353406709) [](ancestors = 353406709,352907086)

Reply:
By placing it on the left-hand-side of a binary operator in a test source.

---
In reply to: [353416906](https://github.com/dotnet/roslyn/pull/40089#discussion_r353416906) [](ancestors = 353416906,353406709,352907086)